tests/src/test/java/org/mozilla/javascript/tests/ApplyTest.java

```diff
@@ -53,4 +53,29 @@ public void applyToCallSetsCorrectFunctionThis() throws Exception {
 
         Utils.assertWithAllModes("b", script);
     }
+
+    @Test
+    public void applyToCallWithActivationCallsCorrectFunction() throws Exception {
+        String script =
+                "function foo(x) {\n"
+                        + "  function inner(s) { return x };\n"
+                        + "  return inner(this);\n"
```

Comment:
my bad this should be 

                        + "  function inner(s) { return s };\n"
                        + "  return inner(x);\n

Reply:
Fixed.